src/Xamarin.Android.Build.Tasks/Tasks/AndroidComputeResPaths.cs

```diff
@@ -133,16 +133,52 @@ public override bool RunTask ()
 					Log.LogCodedError ("XA2001", file: item.ItemSpec, lineNumber: 0, message: Properties.Resources.XA2001, item.ItemSpec);
 					continue;
 				}
+				if (intermediateFiles.TryGetValue (dest, out ITaskItem conflict)) {
+					string conflictItemSpec = conflict.GetMetadata ("OriginalItemSpec");
+					string conflictPath = Path.GetFullPath (conflictItemSpec);
+					string path = Path.GetFullPath (item.ItemSpec);
+					if (string.Compare (conflictPath, path, StringComparison.Ordinal) == 0) {
+						Log.LogCodedWarning (
+							"XA1029",
+							file: conflictItemSpec,
+							lineNumber: 0,
+							message: Properties.Resources.XA1029,
+							conflictItemSpec,
+							conflict.GetMetadata ("LogicalName"),
+							rel);
+						continue;
+					}
+					if (string.Compare (Path.ChangeExtension (conflictPath, ""), Path.ChangeExtension (path, ""), StringComparison.OrdinalIgnoreCase) == 0) {
+						Log.LogCodedError (
+							"XA1030",
+							file: conflictItemSpec,
+							lineNumber: 0,
+							message: Properties.Resources.XA1030,
+							conflictItemSpec,
+							item.ItemSpec);
+						return false;
+					}
+					Log.LogCodedError (
+						"XA1031",
+						file: conflictItemSpec,
+						lineNumber: 0,
+						message: Properties.Resources.XA1031,
+						conflictItemSpec,
+						conflict.GetMetadata ("LogicalName"),
+						item.ItemSpec,
+						rel);
+					return false;
+				}
 				var newItem = new TaskItem (dest);
 				newItem.SetMetadata ("LogicalName", rel);
 				newItem.SetMetadata ("_FlatFile", Monodroid.AndroidResource.CalculateAapt2FlatArchiveFileName (dest));
 				newItem.SetMetadata ("_ArchiveDirectory", AndroidLibraryFlatFilesDirectory);
 				item.CopyMetadataTo (newItem);
-				intermediateFiles.Add (newItem);
+				intermediateFiles.Add (dest, newItem);
 				resolvedFiles.Add (item);
 			}
 
-			IntermediateFiles = intermediateFiles.ToArray ();
+			IntermediateFiles = intermediateFiles.Values.ToArray ();
```

Comment:
@dellis1972 would the ordering of `@(AndroidResource)` files matter? Before the ordering was preserved from the ordering in the `.csproj` file, and this will change them to a random-ish order.

Reply:
It shouldn't matter for those within the app project. The ordering only really matters when you do the final "link" stage. That is where the order of app resources vs library resources really matters.